From 18d1e7aa169b811f975a102e84d21c647ed6186a Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Thu, 19 Mar 2020 20:08:43 +0800
Subject: [PATCH] drm/panel: simple: Introduce CONFIG_DRM_PANEL_SIMPLE_OF_ONLY

Only support panels which all parameters from device tree.

Save boot time about 2ms on rk3126-bnd-d708 board.

Change-Id: Ie075ff5c84c1807029873555f58bd13d41350abf
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
---
 drivers/gpu/drm/panel/Kconfig        | 6 ++++++
 drivers/gpu/drm/panel/panel-simple.c | 4 ++++
 2 files changed, 10 insertions(+)

diff --git a/drivers/gpu/drm/panel/Kconfig b/drivers/gpu/drm/panel/Kconfig
index 6020c30a33b3..3c77449aaba8 100644
--- a/drivers/gpu/drm/panel/Kconfig
+++ b/drivers/gpu/drm/panel/Kconfig
@@ -38,6 +38,12 @@ config DRM_PANEL_SIMPLE
 	  that it can be automatically turned off when the panel goes into a
 	  low power state.
 
+config DRM_PANEL_SIMPLE_OF_ONLY
+	bool "compatible is simple-panel or simple-panel-dsi only"
+	depends on DRM_PANEL_SIMPLE
+	help
+	  Only support panels which all parameters from device tree.
+
 config DRM_PANEL_ILITEK_IL9322
 	tristate "Ilitek ILI9322 320x240 QVGA panels"
 	depends on OF && SPI
diff --git a/drivers/gpu/drm/panel/panel-simple.c b/drivers/gpu/drm/panel/panel-simple.c
index 369e2f8d87ae..0e424f51ee57 100644
--- a/drivers/gpu/drm/panel/panel-simple.c
+++ b/drivers/gpu/drm/panel/panel-simple.c
@@ -2800,6 +2800,7 @@ static const struct of_device_id platform_of_match[] = {
 	{
 		.compatible = "simple-panel",
 		.data = NULL,
+#ifndef CONFIG_DRM_PANEL_SIMPLE_OF_ONLY
 	}, {
 		.compatible = "ampire,am-480272h3tmqw-t01h",
 		.data = &ampire_am_480272h3tmqw_t01h,
@@ -3052,6 +3053,7 @@ static const struct of_device_id platform_of_match[] = {
 	}, {
 		.compatible = "winstar,wf35ltiacd",
 		.data = &winstar_wf35ltiacd,
+#endif /* !CONFIG_DRM_PANEL_SIMPLE_OF_ONLY */
 	}, {
 		/* sentinel */
 	}
@@ -3328,6 +3330,7 @@ static const struct of_device_id dsi_of_match[] = {
 	{
 		.compatible = "simple-panel-dsi",
 		.data = NULL,
+#ifndef CONFIG_DRM_PANEL_SIMPLE_OF_ONLY
 	}, {
 		.compatible = "auo,b080uan01",
 		.data = &auo_b080uan01
@@ -3343,6 +3346,7 @@ static const struct of_device_id dsi_of_match[] = {
 	}, {
 		.compatible = "panasonic,vvx10f004b00",
 		.data = &panasonic_vvx10f004b00
+#endif /* !CONFIG_DRM_PANEL_SIMPLE_OF_ONLY */
 	}, {
 		/* sentinel */
 	}
-- 
2.35.3

